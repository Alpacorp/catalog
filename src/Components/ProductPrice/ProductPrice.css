.product-info .product-price {
  display: flex;
}

.product-info .product-price h3 {
  margin: 0 15px 0 0;
}
