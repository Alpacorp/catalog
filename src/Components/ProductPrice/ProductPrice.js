import PropTypes from "prop-types";
import Price from "../Price/Price";
import PriceDiscount from "../PriceDiscount/PriceDiscount";
import "./ProductPrice.css";

const ProductPrice = ({ discount, price, formatPrice }) => {
  return (
    <div className="product-price">
      {discount ? (
        <>
          <PriceDiscount price={price} formatPrice={formatPrice} />
          <Price price={discount} formatPrice={formatPrice} />
        </>
      ) : (
        <Price price={price} formatPrice={formatPrice} />
      )}
    </div>
  );
};

ProductPrice.propTypes = {
  discount: PropTypes.number,
  price: PropTypes.number,
  formatPrice: PropTypes.object,
};

export default ProductPrice;
