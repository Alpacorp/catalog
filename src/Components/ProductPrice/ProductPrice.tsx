import Price from "../Price/Price";
import PriceDiscount from "../PriceDiscount/PriceDiscount";
import "./ProductPrice.css";

interface ProductPriceProps {
  discount?: number;
  price: number;
  formatPrice?: any;
}

const ProductPrice = ({ discount, price, formatPrice }: ProductPriceProps) => {
  return (
    <div data-testid="test-div-product-price" className="product-price">
      {discount ? (
        <>
          <PriceDiscount price={price} formatPrice={formatPrice} />
          <Price price={discount} formatPrice={formatPrice} />
        </>
      ) : (
        <Price price={price} formatPrice={formatPrice} />
      )}
    </div>
  );
};

export default ProductPrice;
