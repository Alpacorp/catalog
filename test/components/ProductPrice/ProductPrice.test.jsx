import { render, screen } from "@testing-library/react";
import ProductPrice from "../../../src/Components/ProductPrice/ProductPrice";

describe("Test on <ProductPrice />", () => {
  const price = 100;
  const formatPrice = {
    style: "currency",
    currency: "COP",
    minimumFractionDigits: 0,
  };

  test("the component tobe match with snapshot", () => {
    render(<ProductPrice price={price} formatPrice={formatPrice} />);
    expect(screen.getByTestId("test-div-product-price")).toMatchSnapshot();
  });
});
