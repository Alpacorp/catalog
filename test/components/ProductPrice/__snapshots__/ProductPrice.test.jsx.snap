// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <ProductPrice /> the component tobe match with snapshot 1`] = `
<div
  class="product-price"
  data-testid="test-div-product-price"
>
  <div>
    <h3
      aria-label="100"
      tabindex="0"
    />
  </div>
</div>
`;
